publish-mqtt/src/main.rs

```diff
@@ -90,29 +69,48 @@ async fn main() -> Result<(), Box<dyn Error + Send + Sync>> {
 
     let local = task::LocalSet::new();
 
-    let bluetooth_handle = local.spawn_local(async move {
-        bluetooth_mainloop(homie).await.unwrap();
-    });
+    // Connect a bluetooth session.
+    let (dbus_handle, bt_session) = MijiaSession::new().await?;
+
+    let bluetooth_handle =
+        local.run_until(async move { bluetooth_mainloop(homie, bt_session).await });
 
     // Poll everything to completion, until the first one bombs out.
-    let res: Result<_, Box<dyn Error + Send + Sync>> = try_join! {
-        // LocalSet finished first. Colour me confused.
-        local.map(|()| Ok(println!("WTF?"))),
+    let res: Result<_, anyhow::Error> = try_join! {
+        // The resource is a task that should be spawned onto a tokio compatible
```

Comment:
Is the 'resource' the `dbus_handle`?

Reply:
yes. resource is just copy-paste from their example before I renamed it. Will fix.